CMakeLists.txt

```diff
@@ -79,6 +79,8 @@ option(ABSL_USE_GOOGLETEST_HEAD
 
 option(ABSL_RUN_TESTS "If ON, Abseil tests will be run." OFF)
 
+option(ABSL_DECLARE_PUBLIC_SYSTEM_HEADERS "If ON, declare public Abseil headers as SYSTEM in order to mute compiler warnings." OFF)
```

Comment:
What's your opinion on activating/disabling this automatically(still an option), by doing something like this?:
```cmake
if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
    # Don't use system include paths in development
    option(ABSL_DECLARE_PUBLIC_SYSTEM_HEADERS "If ON, declare public Abseil headers as SYSTEM in order to mute compiler warnings." OFF)
else()
    # CMakeLists.txt was included as a subdirectory - disable warnings for the user
    option(ABSL_DECLARE_PUBLIC_SYSTEM_HEADERS "If ON, declare public Abseil headers as SYSTEM in order to mute compiler warnings." ON)
endif()
```

Reply:
Nice idea! To avoid having `option` twice  I would set a variable in the if block (`ABSL_IS_SUBDIRECTORY`?) and use it as the default value in `option`.

Reply:
and adding it inside the block here ?
https://github.com/abseil/abseil-cpp/blob/e96ae2203b80d5ae2e0b7abe0c77b722b224b16d/CMakeLists.txt#L35-L41

Reply:
Done.

Reply:
@Mizux can this be moved forward?